app/code/core/Mage/Catalog/Model/Product.php

```diff
@@ -694,6 +696,32 @@ public function getAttributes($groupId = null, $skipSuper = false)
         return $attributes;
     }
 
+    /**
+     * @return Mage_CatalogInventory_Model_Stock_Item
+     */
+    public function getStockItem()
+    {
+        return $this->_stockItem;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasStockItem()
+    {
+        return !!$this->_stockItem;
```

Comment:
why there is a double negation?

Reply:
it's a cast to boolean as far as I've read

Reply:
I never used it but it is the equivalent ```(bool) $variables```. Maybe the code with this variant !! should be modified to be more explicit for most and not to be able to search the Internet for the "double not" operator in PHP. A longer discussion exists here https://stackoverflow.com/questions/2127260/double-not-operator-in-php. Some people consider using it "fancy".

Reply:
Sorry, this is a habit from writing C.. In PHP probably better to just use (bool) but they are functionally equivalent.